drafts/004-committee-data.md

```diff
@@ -0,0 +1,92 @@
+
+|                    |            |
+|--------------------|------------|
+| **Author(s)**      | @jamesturk |
+| **Implementer(s)** | TODO |
+| **Status**         |   Draft    |
+| **Draft PR(s)**    | https://github.com/openstates/enhancement-proposals/pull/TBD |
+| **Approval PR(s)** | https://github.com/openstates/enhancement-proposals/pull/TBD |
+| **Created**        | 2021-03-31 |
+| **Updated**        | 2021-03-31 |
+
+---
+
+## Abstract
+
+Open States provided committee membership data from 2011-2018, but ceased to when it was no longer viable given the project's funding and staffing situations.  This proposal would restore committee membership data via a hybrid scraped/manual process with a focus on maintaining up-to-date current membership of committees.
+
+## Specification
+
+### Data Model Changes
+
+The data model Open States inherited from OCD/Popolo is still a part of the core OS models and will be renewed to support this work.
+
+As part of this proposal, several small changes will be made to make working with this data slightly easier:
+
+- `OrganizationLink` and `OrganizationSource` will be migrated to inline JSON fields on the `Organization` model.
+- `Organization` will also gain an `other_names` JSON field.
+
+These changes can be implemented in a way that does not impact any of Open States' public APIs.
+
+
+### Scraping
+
+Right now we have 45 committee scrapers in openstates-scrapers that have not been run in years.  A quick check of 10 of them showed that the vast majority do not run as-is.
+
+That fact combined with the limitations in the old pupa import path for Organizations points towards us adopting a different method that allows easier augmentation of scraped data with manual.  To this end, we will port/rewrite committee scrapers to a new format similar to the direction that new people scrapers are heading.
+
+The proposed path looks like this:
+
+- The committee portions of openstates-people will be restored:
+	- each jurisdiction will again have a committees directory
+	- lint_yaml.py will be renamed lint_people.py
+	- a new lint_committees.py will be created
+	- the committee portion of to_database.py will be restored
+	- the committee portion of to_csv.py will be restored
+	- documentation and CI processes will be updated to include committee data
+- Committee scrapers will be added to `openstates-people` and be written in a way that yields compatible YAML.
+	- Existing committee scrapers will be kept in openstates-scrapers until that state has been ported to openstates-people.
+	- new committee scrapers will be run regularly with changes to the YAML submitted as PRs (manually at first, but likely the subject of future automation)
+
+### Manual Data
+
+Initially we'll be focused on manual data that is not impacted by the scrapers, such as alternate names and links.  As such the tool that merges scraper output with the YAML will be focused on replacing member names with those found by the most recent scrape.
```

Comment:
Would there be any plans to drop names from this dataset entirely? From my previous work with the committee data, it seemed duplicative to have names in both committee files and legislator files when there were OCD IDs. (Although I understand that increases the standalone value of those YAML files.)

Reply:
This is a good point that I’d forgotten to address directly, thanks for raising it.  I think it’ll take some thinking about the different ways this data will be curated.

There are four options that are possible, none seem perfect:

• 1. Names only
    • Pushes all reconciliation work to the import, which isn’t what we want to do.
• 2. IDs only
    • Con: Much less user-friendly to edit without tools.
• 3. Names & IDs both required
    • Con: can have name/ID mismatches
    • Redundant if linting enforces the issue.
• 4. Name or ID required
    • Actually a few options within this one that could be considered:
        • a. OK to specify name only unless ambiguous
        • b. ID always required, names are additional context if present but not used
        • c. require ID to match, but allow name in cases where it can’t be matched yet


I think we can rule out 1 and 3 as there aren’t advantages to them.

In thinking about how the scrapers work, I think 4a or 4c might be my top choice, but curious what you think.

Reply:
I think I'd probably prefer 4c. At least from my perspective, the value of the committee data comes from being able to link it to individual legislators for contact info/votes/etc. Given how official names can change around (even sometimes during the same session), it would make more sense to me to prefer an immutable identifier when at all possible.

Reply:
I agree, I updated the manual data section to mention that this is the plan.